idx/cassandra/cassandra.go

```diff
@@ -196,17 +205,20 @@ func (c *CasIdx) Add(data *schema.MetricData) error {
 			return err
 		}
 	}
-	if inMemory {
+
+	if inMemory && existing.Partition != partition {
 		oldest := time.Now().Add(-1 * updateInterval).Add(-1 * time.Duration(rand.Int63n(updateInterval.Nanoseconds()*int64(updateFuzzyness*100))))
 		if existing.LastUpdate < oldest.Unix() {
 			log.Debug("cassandra-idx def hasnt been seem for a while, updating index.")
-			existing.LastUpdate = data.Time
-			c.MemoryIdx.AddDef(&existing)
-			c.writeQueue <- writeReq{recvTime: time.Now(), def: &existing}
+			def := schema.MetricDefinitionFromMetricData(data)
+			def.Partition = partition
+			c.MemoryIdx.AddDef(def)
+			c.writeQueue <- writeReq{recvTime: time.Now(), def: def}
```

Comment:
yeah, that should be if "existing.Partition == partition" so that if the partition hasnt changed we update only every ~updateIterval.  But if the partition has changed we treat it like a new metric which will update it immediately.